### ISSUE TITLE
Query: invalid sql for query projecting a sql function (e.g ROUND) and ordering by it twice

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFrameworkCore#12830]
query:

```cs
                var query = ctx.Orders.Where(o => o.OrderID < 10250).Select(
                    o => new
                    {
                        A = Math.Truncate((double)o.OrderID)
                    }).OrderBy(ee => ee.A).OrderBy(ee => ee.A);

                var result = query.ToList();
```

sql:

```sql
SELECT ROUND(CAST([o].[OrderID] AS float), 0, 1) AS [A]
FROM [Orders] AS [o]
WHERE [o].[OrderID] < 10250
ORDER BY ROUND(CAST([o].[OrderID] AS float), 0, 1), ROUND(CAST([o].[OrderID] AS float), 0, 1)
```

Exception:
A column has been specified more than once in the order by list. Columns in the order by list must be unique.

problem is that when trying to add new entry to order by list, we compare it with existing ones using  `ExpressionEqualityComparer`. The comparer doesn't handle SqlFunctions correctly.

**Comment by maumar** [comment]
problem is that SqlFunctionExpression equals is using SequenceEqual to compare its arguments, which is using a default comparer, rather than our comparer that knows how to handle custom nodes

**Comment by maumar** [comment]
fixed in fbe4c073dcb71c51b905ea6c2025ca30b5839ace
